lib/compress/zstd_compress.c

```diff
@@ -2457,54 +2457,46 @@ static void ZSTD_confirmRepcodesAndEntropyTables(ZSTD_CCtx* zc)
     zc->blockState.nextCBlock = tmp;
 }
 
-static size_t ZSTD_compressBlock_targetCBlockSize_init(ZSTD_CCtx* zc,
-                               void* dst, size_t dstCapacity,
-                               const void* src, size_t srcSize,
-                               U32 lastBlock)
-{
-    const size_t bss = ZSTD_buildSeqStore(zc, src, srcSize);
-    FORWARD_IF_ERROR(bss);
-    if (bss == ZSTDbss_compress)
-        return ZSTD_compressSuperBlock(zc, dst, dstCapacity, lastBlock);
-    return 0;
-}
-
-static void ZSTD_compressBlock_targetCBlockSize_end(ZSTD_CCtx* zc)
-{
-    if (zc->blockState.prevCBlock->entropy.fse.offcode_repeatMode == FSE_repeat_valid)
-        zc->blockState.prevCBlock->entropy.fse.offcode_repeatMode = FSE_repeat_check;
-}
-
 static size_t ZSTD_compressBlock_targetCBlockSize_body(ZSTD_CCtx* zc,
-                               size_t cSize, void* dst, size_t dstCapacity,
+                               const size_t bss, void* dst, size_t dstCapacity,
                                const void* src, size_t srcSize,
                                U32 lastBlock)
 {
-    /* Superblock compression was successful */
-    if (cSize != 0) {
-        ZSTD_confirmRepcodesAndEntropyTables(zc);
-        return cSize;
+    /* Attempt superblock compression and return early if successful */
+    {
+        if (bss == ZSTDbss_compress) {
+            size_t cSize = ZSTD_compressSuperBlock(zc, dst, dstCapacity, lastBlock);
```

Comment:
minor : `size_t const cSize =`.
Try to `const` any variable that's not modified after its creation.

Sure, it's a few more keystrokes, but it offers a useful signal to the reader : 
this variable is not going to change, so I can "trust" its value.

Bonus, and probably as important : when this rule is applied consistently, now when a variable is _not_ `const`, it's supposed to mean that this variable _is_ going to change during its lifetime, so the reader can focus its attention on it. This is an important signal because mutable variables are generally the ones where bugs happen.

As usual, the goal is to simplify the reader's understanding.

There are several other variables below that can also be made `const`.